api/services/lau.ts

```diff
@@ -0,0 +1,231 @@
+  SPECIFIC: 'specific',
+  CHALLENGED: 'challenged'
+};
+
+const CHALLENGED_REASON = {
+  0: 'The cases or parties are linked to the case I am working on',
+  1: 'To determine if the case needs to be consolidated',
+  2: 'To consider an order for transfer',
+  3: 'Other reason'
+};
+
+export type AccessLogContainer = {
+  accessLog: AccessLog
+}
+
+export type AccessLog = {
+  requestType: string;
+  caseRef: string;
+  userId: string;
+  action: string;
+  timestamp: string;
+  reason: string;
+  requestStartTimestamp: string;
+  requestEndTimestamp: string;
+}
+
+export interface detailObject {
+  reason?: number;
+  otherReason?: string;
+  specificReason?: string;
+  caseReference?: string;
+}
+
+export function setFeatureSpecificChallengedAccessEnabled(value: boolean) {
+  featureSpecificChallengedAccessEnabled = value;
+}
+
+export async function logAccessRequest(req: EnhancedRequest, isRequest: boolean) {
+  if (featureSpecificChallengedAccessEnabled) {
+    if (isRequest) {
+      return await createAccessLogFromRequest(req);
+    }
+    return await createAccessLogFromDecision(req);
+  }
+}
+
+/**
+ * Create access log from request. If it is Challenged access request - it is
+ * automatically approved, and we set action to APPROVED. Request itself is in
+ * the form of AM role assignment request, e.g:
+ * {
+ *   roleRequest: {
+ *     assignerId: '7a51b87d-36c6-47fa-b9ef-4257af8001c5',
+ *     process: 'specific-access',
+ *     reference: '1708968484691010/specific-access-judiciary/7a51b87d-36c6-47fa-b9ef-4257af8001c5',
+ *     replaceExisting: true
+ *   },
+ *   requestedRoles: [
+ *     {
+ *       actorIdType: 'IDAM',
+ *       actorId: '7a51b87d-36c6-47fa-b9ef-4257af8001c5',
+ *       roleType: 'CASE',
+ *       roleName: 'specific-access-requested',
+ *       classification: 'PRIVATE',
+ *       roleCategory: 'JUDICIAL',
+ *       grantType: 'BASIC',
+ *       beginTime: null,
+ *       endTime: '2024-10-23T13:30:24.196Z',
+ *       attributes: {
+ *         caseId: '1708968484691010',
+ *         isNew: true,
+ *         accessReason: '{"specificReason":"my own reason justification"}',
+ *         requestedRole: 'specific-access-judiciary',
+ *         specificAccessReason: 'I have my own reasons'
+ *       },
+ *       notes: [
+ *         {
+ *           userId: '7a51b87d-36c6-47fa-b9ef-4257af8001c5',
+ *           time: '2024-09-24T10:09:07.554Z',
+ *           comment: '{"specificReason":"I have my own reasons"}'
+ *         }
+ *       ],
+ *       readOnly: true
+ *     }
+ *   ]
+ * }
+ * @param req
+ */
+async function createAccessLogFromRequest(req: EnhancedRequest) {
+  const process = req.body.roleRequest.process;
+  try {
+    const detail: detailObject = JSON.parse(req.body.requestedRoles[0].notes[0].comment);
+    const accessLog: AccessLog = {
+      requestType: (process === 'specific-access') ? REQUEST_TYPE.SPECIFIC : REQUEST_TYPE.CHALLENGED,
+      caseRef: req.body.requestedRoles[0].attributes.caseId,
```

Comment:
For resilience, I think it would be good to use optional chaining in all these attribute accesses. As it stands you'll get no log if any of the data is malformed - a partial log is better than none I think. So:
reg?.body?.requestedRoles[0]?.attributes?.caseId 
etc

Reply:
Addressed, though I didn't do for `req.body`, if we didn't have those, there wouldn't be anything to pass.